src/directives/exportfeatures.js

```diff
@@ -0,0 +1,214 @@
+goog.provide('ngeo.ExportfeaturesController');
+goog.provide('ngeo.exportfeaturesDirective');
+
+goog.require('ngeo');
+
+
+/**
+ * Directive used to export vector features in different types of format.
+ * To configure which formats to use, define the `ngeoExportFeatureFormats`
+ * constant, as such:
+ *
+ *     app.module.constant('ngeoExportFeatureFormats', [
+ *         ngeo.FeatureHelper.FormatType.KML,
+ *         ngeo.FeatureHelper.FormatType.GPX
+ *     ]);
+ *
+ * Example:
+ *
+ *     <button
+ *       ngeo-exportfeatures
+ *       ngeo-exportfeatures-features="ctrl.features"
+ *       class="btn btn-link">Export</button>
+ *
+ * @htmlAttribute {ol.Collection.<ol.Feature>} ngeo-exportfeatures-features The
+ *     features to export
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname ngeoExportfeatures
+ */
+ngeo.exportfeaturesDirective = function() {
+  return {
+    controller: 'ngeoExportfeaturesController',
+    scope: true,
+    bindToController: {
+      'features': '=ngeoExportfeaturesFeatures'
+    },
+    controllerAs: 'efCtrl'
+  };
+};
+
+
+ngeo.module.directive('ngeoExportfeatures', ngeo.exportfeaturesDirective);
+
+
+/**
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.$injector} $injector Main injector.
+ * @param {!angular.Scope} $scope Angular scope.
+ * @param {ngeo.FeatureHelper} ngeoFeatureHelper Ngeo feature helper service.
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname ngeoExportfeaturesController
+ */
+ngeo.ExportfeaturesController = function($element, $injector, $scope,
+    ngeoFeatureHelper) {
+
+  /**
+   * @type {ol.Collection.<ol.Feature>}
+   * @private
+   */
+  this.features;
+
+  /**
+   * @type {angular.JQLite}
+   * @private
+   */
+  this.element_ = $element;
+
+  var uid = goog.getUid(this);
+  var id = ['ngeo-exportfeature', uid].join('-');
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.id_ = id;
+
+  /**
+   * @type {ngeo.FeatureHelper}
+   * @private
+   */
+  this.featureHelper_ = ngeoFeatureHelper;
+
+  var formats;
+  if ($injector.has('ngeoExportFeatureFormats')) {
+    formats = $injector.get('ngeoExportFeatureFormats');
+  } else {
+    formats = [ngeo.FeatureHelper.FormatType.KML];
+  }
+
+  /**
+   * @type {?jQuery}
+   * @private
+   */
+  this.menu_ = null;
+
+  /**
+   * @type {Array.<jQuery>}
+   * @private
+   */
+  this.items_ = [];
+
+  // build the drop-down menu and items of there's more than one format
+  if (formats.length > 1) {
```

Comment:
I don't have the reflexes to use Angular as it's not always obvious for me to use it when it can be.  In this case, I don't see how it can be used, but that's just a lack of understanding on my part. Would you please explain this to me ???? :)